backend/app/core/security.py

```diff
@@ -41,6 +102,73 @@ def get_token_data(token: str = Cookie(None)) -> TokenData:
         )
 
 
+def generate_reset_token(email: str) -> str:
+    expiration_time = datetime.utcnow() + timedelta(minutes=TOKEN_EXPIRE_MINUTES)
+    payload = {"email": email, "exp": expiration_time}
+    token = jwt.encode(payload, RESET_PASSWORD_KEY, algorithm=ALGORITHM)
+    return token
+
+
+async def send_reset_email(email: EmailStr, token: str) -> JSONResponse:
+    html = f"""
+    <html>
+      <head></head>
+      <body>
+        <p>Hello,</p>
+        <p>You have requested to reset your password. Please click the link below to reset your password:</p>
+        <p><a style="background-color: #008CBA; border: none; color: white; padding: 15px 32px; text-align: center; text-decoration: none; display: inline-block; font-size: 16px; margin: 4px 2px; cursor: pointer;"
+        href="http://localhost:3000/reset-password/{email}/{token}">Reset Password</a></p>
+        <p>If you did not request a password reset, please ignore this email.</p>
+      </body>
+    </html>
+    """
+
```

Comment:
Add this email to the template folder as HTML file.

Reply:
Created separate template files for each html code snippet : then used the templates wherever needed:

```
context = {"verify_url": verify_url, "email": email, "token": token}
    html = load_template("verification_email.html", context)
```